crates/fast_compute/src/layer.rs

```diff
@@ -57,6 +71,88 @@ impl<T: TowerFamily, P: PackedTop<T>> Default for FastCpuLayer<T, P> {
 	}
 }
 
+/// Extrapolate_line taking advantage of transforming to the byte-sliced representation and back.
+/// This is used for the PackedBinaryField1x128b, PackedBinaryField2x128b, and
+/// PackedBinaryField4x128b types only.
```

Comment:
Please explain in the comment why this must be a macro instead of a generic function.

Reply:
While writing the comment pointing to the places which cannot be expressed with generic code I've eventually found a workaround for each of those and managed to express it with an ugly but generic function instead of a macro.